src/vm/arm64/asmhelpers.asm

```diff
@@ -1106,9 +1106,9 @@ MainLoop
         
 Success         
         ldr     x13, =g_dispatch_cache_chain_success_counter
-        ldr     x9, [x13]
-        subs    x9, x9, #1
-        str     x9, [x13]
+        ldr     x16, [x13]
+        subs    x16, x16, #1
+        str     x16, [x13]
         blt     Promote
 
         ldr     x16, [x9, #16]    ; get the ImplTarget
```

Comment:
can you please fix this function to use macros instead of directly using byte offsets 16 and 24 at line 1096. We also add assertions in asmconstants.h so they match the C structure.